### PR TITLE
[DOCS-6951] Update trace and span ID page to reflect 128-bit IDs

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

All Datadog tracing libraries now generate 128-bit IDs by default. Update the Trace and Span ID Formats page to reflect the new paradigm.

- By default, all tracing libraries generate 128-bit trace IDs and 64-bit span IDs. 
- Datadog accepts 128-bit or 64-bit trace IDs, and 64-bit span IDs.
- Trace ID range should be 0, 2^64-1 for all languages (differing from span IDs).

Edit: Removed ID ranges from the new sections after discussing with js team.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/tracing/guide/span_and_trace_id_format.md [hunk 1]
```diff
@@ -3,12 +3,54 @@ title: Trace and Span ID Formats
...
+This page details Datadog tracing library support for trace and span IDs.
+
+- **Generated IDs**: By default, all tracing libraries generate 128-bit trace IDs and 64-bit span IDs.  
+- **Accepted IDs**: Datadog accepts 128-bit or 64-bit trace IDs, and 64-bit span IDs.
+
+## 128-bit trace IDs
+
+128-bit trace IDs are generated and accepted by default in the latest versions of Datadog tracing libraries.
+
+| Language   | Generated IDs             | Valid Accepted 128-bit int IDs |
```
Comment (alai97) [inline content/en/tracing/guide/span_and_trace_id_format.md:23]:
```suggestion
| Language   | Generated IDs             | Valid, Accepted 128-bit int IDs |
```

File: content/en/tracing/guide/span_and_trace_id_format.md [hunk 2]
```diff
@@ -3,12 +3,54 @@ title: Trace and Span ID Formats
...
+| C++        | Unsigned [0, $2^128-1$]   |  Unsigned                      |
+
+## 64-bit trace and span IDs
+
+### Trace IDs
 
-Generally, the libraries generate IDs that are 64-bit unsigned integers. Specifics and exceptions noted below:
+Trace IDs are generated as 128-bit by default, and they are accepted as either 128-bit or 64-bit integers. To generate 64-bit trace IDs, set the environment variable `DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED` to `false`.
+
+| Language   | Generated IDs            | Valid Accepted 64-bit int IDs |
```
Comment (alai97) [inline content/en/tracing/guide/span_and_trace_id_format.md:40]:
```suggestion
| Language   | Generated IDs            | Valid, Accepted 64-bit int IDs |
```

File: content/en/tracing/guide/span_and_trace_id_format.md [hunk 3]
```diff
@@ -3,12 +3,54 @@ title: Trace and Span ID Formats
...
+| Ruby       | Unsigned  [$0, 2^64-1$]   | Unsigned                      |
+| .NET       | Unsigned  [$0, 2^64-1$]   | Unsigned                      |
+| PHP        | Unsigned  [$0, 2^64-1$]   | Unsigned                      |
+| C++        | Unsigned  [$0, 2^64-1$]   | Unsigned                      |
+
+### Span IDs
+
+Span IDs are limited to 64-bits in Datadog.
 
 | Language   | Generated IDs            | Valid Accepted 64-bit int IDs |
```
Comment (alai97) [inline content/en/tracing/guide/span_and_trace_id_format.md:55]:
```suggestion
| Language   | Generated IDs            | Valid, Accepted 64-bit int IDs |
```

File: content/en/tracing/guide/send_traces_to_agent_by_api.md [hunk 4]
```diff
@@ -56,7 +58,7 @@ and each span is a dictionary with a `trace_id`, `span_id`, `resource` and so on
 | `service`    | string  | The service you are tracing. The service name must not be longer than 100 characters. |
 | `span_id`    | int64   | The span integer (64-bit unsigned) ID. |
 | `start`      | int64   | The start time of the request in nanoseconds from the UNIX epoch. |
-| `trace_id`   | int64   | The unique integer (64-bit unsigned) ID of the trace containing this span. |
+| `trace_id`   | int64 or int128   | The unique integer (64-bit unsigned or 128-bit unsigned) ID of the trace containing this span. |
```
Comment (dinooliva) [inline content/en/tracing/guide/send_traces_to_agent_by_api.md:61]: It depends a bit on what `trace_id` means here.

If we are talking about the logical `trace_id` field then this is fine.

However, our msgpack and protobuf representations of trace data contain a uint64 trace_id field, which contains the lower-order 64-bits of the trace id - if that's what's being referred to, customers will find this confusing. They can use the lower-order 64 bits of the trace id to fill in this field and we will match it up to the 128-bit trace id in the backend.

Comment (brett0000FF) [reply content/en/tracing/guide/send_traces_to_agent_by_api.md:61]: Thanks for taking a look. In my opinion, we should focus on the logical `trace_id` field. As you mentioned before as well, we plan to change this representation eventually so it'd be best not to expose these details.

**Review by dinooliva — APPROVED** [review]

**Review by rochdev — COMMENTED** [review]
Generally speaking, I think this change focuses too much on implementation details. Couldn't we just describe what is the expected subset that works everywhere? Especially since this will otherwise be a moving target as languages make improvements to their handlings of IDs.

**Pull request merged**